js/apps/system/_admin/aardvark/APP/react/src/views/collections/indices/addIndex/AddIndexForm.tsx

```diff
@@ -0,0 +1,68 @@
+import { Box, Stack } from "@chakra-ui/react";
+import React, { useState } from "react";
+import SingleSelect from "../../../../components/select/SingleSelect";
+import { useCollectionIndicesContext } from "../CollectionIndicesContext";
+import { FulltextIndexForm } from "./FulltextIndexForm";
+import { GeoIndexForm } from "./GeoIndexForm";
+import { InfoTooltip } from "./InfoTooltip";
+import { PersistentIndexForm } from "./PersistentIndexForm";
+import { TTLIndexForm } from "./TTLIndexForm";
+import { ZKDIndexForm } from "./ZKDIndexForm";
+
+export const AddIndexForm = ({ onClose }: { onClose: () => void }) => {
+  const { indexTypeOptions } = useCollectionIndicesContext();
+  const [indexType, setIndexType] = useState(indexTypeOptions?.[0].value || "");
+  let tooltipText = "Type of index to create.";
+  if (!indexTypeOptions?.find(option => option.value === "hash")) {
+    tooltipText = `${tooltipText} Please note that for the RocksDB engine the index types "hash", "skiplist' and "persistent" are identical, so that they are not offered seperately here.`;
+  }
+  return (
+    <Box width="100%" padding="4">
+      <Stack padding="4" background="white" spacing="4">
+        <Box
+          display={"grid"}
+          gridTemplateColumns={"200px 1fr 40px"}
+          rowGap="5"
+          columnGap="3"
+          maxWidth="500px"
+        >
+          <Box fontSize={"lg"}>Add new index</Box>
+          <SingleSelect
+            defaultValue={indexTypeOptions?.[0]}
+            options={indexTypeOptions}
+            onChange={value => {
+              setIndexType((value as any).value);
```

Comment:
I am facing one issue while reading similar constructs. How can I guess that the key `.value` will be available for the param `value` in the `onChange`. Would it make sense to assert on it's actual type? This way we can know what keys can be used? I get the `any` assertion but hoping to find a slightly more readable approach 🙂

Reply:
found a fix for this using generics!